### PR TITLE
examples: Add Torus cluster example and tutorial

### PR DESCRIPTION
**Pull request by dghubble**
rel #227 cc @barakmich @gyuho

### CODE REVIEW

**Comment by gyuho** [comment]
etcd part, LGTM. Thanks!

File: examples/ignition/torus.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      enable: true
+      contents: |
+        [Unit]
+        Description=etcd3
+        Conflicts=etcd2.service
+        After=docker.service
+        Requires=docker.service
+        [Service]
+        ExecStartPre=/usr/bin/mkdir -p /var/lib/etcd3
+        ExecStart=/usr/bin/docker run --net=host quay.io/coreos/etcd:v3.0.0-beta.0 -name {{.etcd_name}} -advertise-client-urls http://{{.ipv4_address}}:2379 -listen-client-urls http://0.0.0.0:2379 --initial-advertise-peer-urls http://{{.ipv4_address}}:2380 --listen-peer-urls http://{{.ipv4_address}}:2380 --initial-cluster {{.etcd_initial_cluster}} -data-dir /var/lib/etcd3
```
Comment (barakmich) [inline examples/ignition/torus.yaml]: These flags can also be environment vars, eg, `ETCD_LISTEN_PEER_URLS=http://{{.ipv4_address}}:2380` -- whichever is cleaner

File: examples/ignition/torus.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        Requires=etcd3.service
+        After=etcd3.service
+        Requires=docker.service
+        After=docker.service
+        [Service]
+        ExecStartPre=/usr/bin/mkdir -p /srv/torus
+        ExecStart=/usr/bin/docker run \
+          --net=host \
+          -v /srv/torus:/data \
+          -e STORAGE_SIZE=1GiB \
```
Comment (barakmich) [inline examples/ignition/torus.yaml]: Fine for the example, but maybe make this a `{{.storage_size}}` to drive it home?

File: examples/ignition/torus.yaml (hunk 1 above)
Comment (dghubble) [inline examples/ignition/torus.yaml]: Done

File: examples/ignition/torus.yaml (hunk 2 above)
Comment (dghubble) [inline examples/ignition/torus.yaml]: `torus_storage_size`

File: examples/ignition/torus.yaml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        ExecStartPre=/usr/bin/mkdir -p /srv/torus
+        ExecStart=/usr/bin/docker run \
+          --net=host \
+          -v /srv/torus:/data \
+          -e STORAGE_SIZE={{.torus_storage_size}} \
+          -e LISTEN_HOST={{.ipv4_address}} \
+          -e ETCD_HOST=127.0.0.1 \
+          -e ETCD_PORT=2379 \
+          -e AUTO_JOIN=1 \
+          -e DEBUG_INIT=1 \
```
Comment (dghubble) [inline examples/ignition/torus.yaml]: I think DEBUG_MKFS was from before release and the correct setting is DEBUG_INIT now

**Comment by dghubble** [comment]
@barakmich Added a Torus specific tutorial, similar to the existing Kubernetes and self-hosted Kubernetes tutorials.

**Comment by barakmich** [comment]
LGTM

**Pull request merged**